examples/3d/3d_viewport_to_world.rs

```diff
@@ -0,0 +1,96 @@
+//! This example demonstrates how to use the `Camera::viewport_to_world` method.
+
+use bevy::{pbr::CascadeShadowConfigBuilder, prelude::*};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system(move_cube)
+        .run();
+}
+
+fn move_cube(
+    camera_query: Query<(&Camera, &GlobalTransform)>,
+    mut cube_query: Query<&mut Transform, With<Cube>>,
+    windows: Query<&Window>,
+    input: Res<Input<MouseButton>>,
+) {
+    let (camera, camera_transform) = camera_query.single();
+    let mut transform = cube_query.single_mut();
+
+    if !input.pressed(MouseButton::Left) {
+        return;
+    }
+
+    let Some(cursor_position) = windows.single().cursor_position() else { return; };
+
+    let Some(ray) = camera.viewport_to_world(camera_transform, cursor_position) else { return; };
+
+    let Some(distance) = ray.intersect_plane(Vec3::ZERO, Vec3::Y) else { return; };
+
+    let point = ray.get_point(distance);
+    transform.translation = point + Vec3::Y * 0.5;
```

Comment:
I'm wondering if this section might be somewhat magical for new users.

- `Vec3::ZERO, Vec3::Y` duplicates the position / normal of the `Plane` we built in `setup`. 
- `Vec3::Y * 0.5` duplicates the offset we defined for the cube in `setup`.

I played around with this a smidge, and I think

`intersect_plane(ground.translation, ground.up())` reads well.

Querying for `Aabb` and placing the cube based on its midpoint feels sort of messy. Maybe a `CUBE_HEIGHT` constant or something would be clearer for the example?

Or possibly just some comments would help.